tmail-backend/webadmin/webadmin-mailbox/src/main/java/com/linagora/tmail/webadmin/archival/InboxArchivalService.java

```diff
@@ -0,0 +1,146 @@
+            .map(Throwing.function(MessageManager::getMailboxEntity))
+            .flatMapMany(getMailboxMessagesMetadata(mailboxSession))
+            .filter(messagesOlderThanArchiveDate(archiveDate))
+            .flatMap(mailboxMessage -> archiveMessage(mailboxMessage, mailboxSession, jmapSettings), ReactorUtils.DEFAULT_CONCURRENCY)
+            .then(Mono.just(Task.Result.COMPLETED))
+            .onErrorResume(e -> {
+                LOGGER.error("Error while archiving INBOX for user {}", username.asString(), e);
+                return Mono.just(Task.Result.PARTIAL);
+            });
+    }
+
+    private FunctionChainer<Mailbox, Flux<MailboxMessage>> getMailboxMessagesMetadata(MailboxSession mailboxSession) {
+        return Throwing.function(mailbox -> mapperFactory.getMessageMapper(mailboxSession)
+            .findInMailboxReactive(mailbox, MessageRange.all(), MessageMapper.FetchType.METADATA, UNLIMITED));
+    }
+
+    private Predicate<MailboxMessage> messagesOlderThanArchiveDate(Date archiveDate) {
+        return mailboxMessage -> mailboxMessage.getSaveDate()
+            .orElse(mailboxMessage.getInternalDate())
+            .before(archiveDate);
+    }
+
+    private Mono<Void> archiveMessage(MailboxMessage mailboxMessage, MailboxSession mailboxSession, JmapSettings jmapSettings) {
+        return switch (jmapSettings.inboxArchivalFormat().toString()) {
+            case "yearly" -> yearlyArchive(mailboxMessage, mailboxSession);
+            case "monthly" -> monthlyArchive(mailboxMessage, mailboxSession);
+            default -> singleArchive(mailboxMessage, mailboxSession);
+        };
+    }
+
+    private Mono<Void> singleArchive(MailboxMessage mailboxMessage, MailboxSession mailboxSession) {
+        return moveMessage(mailboxMessage, mailboxSession, MailboxPath.forUser(mailboxSession.getUser(), DefaultMailboxes.ARCHIVE));
+    }
+
+    private Mono<Void> yearlyArchive(MailboxMessage mailboxMessage, MailboxSession mailboxSession) {
+        int savedYear = mailboxMessage.getSaveDate()
+            .orElse(mailboxMessage.getInternalDate())
+            .toInstant()
+            .atZone(ZoneId.of("UTC"))
+            .toLocalDate()
+            .getYear();
+        MailboxPath yearlyArchiveSubMailbox = MailboxPath.forUser(mailboxSession.getUser(), String.format("Archive.%d", savedYear));
+
+        return moveMessage(mailboxMessage, mailboxSession, yearlyArchiveSubMailbox);
+    }
+
+    private Mono<Void> monthlyArchive(MailboxMessage mailboxMessage, MailboxSession mailboxSession) {
+        LocalDate savedDate = mailboxMessage.getSaveDate()
+            .orElse(mailboxMessage.getInternalDate())
+            .toInstant()
+            .atZone(ZoneId.of("UTC"))
+            .toLocalDate();
+        MailboxPath monthlyArchiveSubMailbox = MailboxPath.forUser(mailboxSession.getUser(),
+            String.format("Archive.%d.%s", savedDate.getYear(), savedDate.getMonth().getDisplayName(TextStyle.FULL, Locale.US)));
```

Comment:
You can't infer the local for month display: `01` is waaaay better than `January` for my Mum for instance.

Let's do this based on the user local setting then? Or maybe front would be able to adjust the display?

Reply:
> Let's do this based on the user local setting then?

I can think about some month display formats that can be done on the backend side:
  - number: 01 02 ... 12
  - full month name: January, February,...
  - Short month name: Jan, Feb,...
 
Less work on the frontend side.

In the case user changes the month display format, we can either:
- Adjust the display on the frontend side
- Or on backend side we listen to the Setting change event and rename the archived mailboxes.

Reply:
The problem is that you cannot infer that the user speaks english!

Reply:
Both solution you propose are OK and would be an enhancement.

Reply:
I think numbers are enough IMO. It talks for everyone despite their language, let's keep it simple? :)

Reply:
Yes. Let's keep it number then, and let frontend side manipulate the display if needed.